include/support.h

```diff
@@ -230,6 +230,10 @@ using FILE_unique_ptr = std::unique_ptr<FILE, FILE_closer>;
 // itself when it goes out of scope
 FILE_unique_ptr make_fopen(const char *fname, const char *mode);
 
+int64_t stdio_size_bytes(FILE* f);
+int64_t stdio_size_kb(FILE* f);
+int64_t stdio_num_sectors(FILE* f);
```

Comment:
Suggest `const FILE*` type, to match all the other `const`s and to assert we're not changing `f`.

Reply:
Actually, just thinking about it, we can't make these const, `fseeko` modifies the file pointer.

And I probably need to save and restore the file position? Either that or make a note that the file position will change. Thoughts?

Reply:
Oh - great catch!
Yes, definitely save and restore the pos in the sunny day scenario.

I guess if any of these sub-steps fails, then all bets are off that the pos will be restored too (maybe not that in the function comment).